### PR TITLE
Add support for medianizing LLO TimestampedStreamValue timestamps

### PR DESCRIPTION
**Pull request by martin-cll**
- Updates the LLO integration test to use a more realistic bridge response. Rather than relying on there being a single `"result"` value which is not true in practice (result can be null), we can explicitly set the result in the merge. Include tests for parsing and medianizing null/missing `providerIndicatedTimeUnix` fields.
- Handle nil timestamp fields when parsing TimestampedStreamValue
- Add `Lax` option to the `median` pipeline task and `FilterNils()` in order to medianize potentially nil timestamp values

### CODE REVIEW

**Review by brunotm — DISMISSED** [review]

File: core/services/llo/observation/observation_context.go [hunk 1]
```diff
@@ -252,8 +253,20 @@ func toUint64(v interface{}) (uint64, error) {
 			return 0, fmt.Errorf("expected positive int64, got: %d", v)
 		}
 		return uint64(v), nil
+	case decimal.Decimal:
```
Comment (martin-cll) [inline core/services/llo/observation/observation_context.go:256]: I'm curious why you added decimal support here, is a decimal value being passed into this function?

Comment (calvwang9) [reply core/services/llo/observation/observation_context.go:256]: yes, the `median` task returns a Decimal type. Since this `toUint64` function already supports conversion of float64 to uint64, I figured adding decimal support here would be the easiest solution

**Review by calvwang9 — APPROVED** [review]

**Review by brunotm — APPROVED** [review]

File: core/services/pipeline/task_params.go [hunk 2]
```diff
@@ -556,6 +556,19 @@ func (s SliceParam) FilterErrors() (SliceParam, int) {
...
 }
 
+func (s SliceParam) FilterNils() (SliceParam, int) {
+	var s2 SliceParam
+	var nils int
+	for _, x := range s {
+		if x == nil {
+			nils++
+		} else {
+			s2 = append(s2, x)
```
Comment (EasterTheBunny) [inline core/services/pipeline/task_params.go:566]: Are the slices expected to be large? This might be inefficient.

**Review by EasterTheBunny — APPROVED** [review]

File: core/services/pipeline/task_params.go (hunk 2 above)
Comment (jmank88) [reply core/services/pipeline/task_params.go:566]: `s2` could be preallocated with cap set to len(s) :shrug:

**Review by jmank88 — APPROVED** [review]

File: core/services/pipeline/task_params.go (hunk 2 above)
Comment (calvwang9) [reply core/services/pipeline/task_params.go:566]: makes sense 👍 the slices handled by this fn wouldnt be too big, and in the interest of getting this pr through to unblock other work i'll defer this optimization to be handled alongside FilterErrors() (exact same implementation) in a future pr
